### PR TITLE
Reenable TestControllerCacheFreshness

### PR DESCRIPTION
**Pull request by xiaolanz**
Another attempt after #4149 was reverted.
adresses #4147

### CODE REVIEW

File: pilot/test/mock/config.go [hunk 1]
```diff
@@ -596,11 +592,13 @@ func CheckCacheFreshness(cache model.ConfigStoreCache, namespace string, t *test
 		t.Error(err)
 	}
 
-	util.Eventually("successfully set done", func() bool {
-		doneMu.Lock()
-		defer doneMu.Unlock()
-		return done
-	}, t)
+	timeout := time.After(10 * time.Second)
```
Comment (mandarjog) [inline pilot/test/mock/config.go]: What was the total timeout before with eventually?

Comment (xiaolanz) [reply pilot/test/mock/config.go]: Total waiting time will be about 1min. But for this test, I don't expect it to be that long. 

I put 60s in controller test. https://github.com/istio/istio/pull/4148/files

**Comment by xiaolanz** [comment]
a  rebase error. now reopen.

**Pull request merged**